truth_set/python/structure/solution-approach/semantic-duplication/services/orderService.py

```diff
@@ -0,0 +1,32 @@
+from datetime import datetime
+
+class OrderService:
+    def __init__(self):
+        self.orders = []
+    
+    def add_order(self, product):
+        new_order = {
+            'order_id': len(self.orders) + 1,
+            'product': product,
+            'order_date': datetime.now()
+        }
+        self.orders.append(new_order)
+        formatted_date = self.format_date(new_order['order_date'], 'DD-MM-YYYY')
+        print(f"Order Added: {new_order['product']}, Order Date: {formatted_date}")
+    
+    def list_orders(self):
+        for order in self.orders:
+            formatted_date = self.format_date(order['order_date'])
+            print(f"Order ID: {order['order_id']}, Product: {order['product']}, Order Date: {formatted_date}")
+    
+    def format_date(self, date, format='YYYY-MM-DD'):
+        year = date.year
+        month = f"{date.month:02d}"
+        day = f"{date.day:02d}"
+        
+        if format == 'MM/DD/YYYY':
+            return f"{month}/{day}/{year}"
+        elif format == 'DD-MM-YYYY':
+            return f"{day}-{month}-{year}"
+        else:
+            return f"{year}-{month}-{day}"
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider reusing existing date formatting utility</b></div>
<div id="fix">

Consider using the existing `format_date` function from `dateUtils.py` instead of duplicating the date formatting logic in `OrderService`. This would improve code maintainability and reduce duplication.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -1,32 +1,27 @@
 from datetime import datetime
 from utils.dateUtils import format_date

 class OrderService:
     def __init__(self):
         self.orders = []

     def add_order(self, product):
         new_order = {
             'order_id': len(self.orders) + 1,
             'product': product,
             'order_date': datetime.now()
         }
         self.orders.append(new_order)
         formatted_date = format_date(new_order['order_date'], 'DD-MM-YYYY')
         print(f"Order Added: {new_order['product']}, Order Date: {formatted_date}")

     def list_orders(self):
         for order in self.orders:
             formatted_date = format_date(order['order_date'])
             print(f"Order ID: {order['order_id']}, Product: {order['product']}, Order Date: {formatted_date}")
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/1#issuecomment-2536353661>#c621b6</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged